### PR TITLE
Physical connections implemented as AST transformation 

### PR DESCRIPTION
**Pull request by erlingrj**
Addresses #616. A simple way of achieving this is to reuse the AfterDelay transformation, just with a Physcial Action instead. This PR does the following:
1. Rename AfterDelayTransformation -> DelayedConnectionTransformation. This takes two paramters `boolean transformAfterDelays` and `transformPhysicalConnections` which enables each of the two different transformations. 
2. Add some extra control logic to transform the physical connection using a generated delay reactor with a physical action instead of a logical one.

The main discussion point is whether we would prefer to make a separate Transformation class for the PhysicalConnections. I initially decided not to, since it is so similar to the AfterDelays. If we decide this is OK, I will make another pass and add documenting comments.

### CODE REVIEW

**Review by lhstrh — COMMENTED** [review]
:+1: This looks pretty clean to me. I don't see the need for a separate class... Thanks for taking on this issue! It has lingered for a long time...

**Review by edwardalee — APPROVED** [review]
LGTM!

File: org.lflang/src/org/lflang/generator/ts/TSGenerator.kt [hunk 1]
```diff
@@ -103,7 +103,7 @@ class TSGenerator(
      */
     override fun doGenerate(resource: Resource, context: LFGeneratorContext) {
         // Register the after delay transformation to be applied by GeneratorBase.
-        registerTransformation(AfterDelayTransformation(TSDelayBodyGenerator, targetTypes, resource))
+        registerTransformation(DelayedConnectionTransformation(TSDelayBodyGenerator, targetTypes, resource, true, false))
```
Comment (cmnrd) [inline org.lflang/src/org/lflang/generator/ts/TSGenerator.kt:106]: Shouldn't we enable the physical connections also in TS?

**Review by cmnrd — COMMENTED** [review]
This looks good indeed. I think this is exactly how physical connections should be handled. Having a separate transformation would only duplicate code. The only comment I have, is that we should probably also enable physical connection in TS and at a test for it.

**Comment by hokeun** [comment]
@byeong-gil kindly agreed to work on this. Thank you!

**Comment by erlingrj** [comment]
> @byeong-gil kindly agreed to work on this. Thank you!

Not sure I understand. This PR is more-or-less done?

File: test/TypeScript/src/PhysicalConnection.lf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    output y: number
+
+    reaction(startup) -> y {= y = 42 =}
+}
+
+reactor Destination {
+    input x: number
+
+    reaction(x) {=
+        let time =util.getElapsedLogicalTime;
```
Comment (lhstrh) [inline test/TypeScript/src/PhysicalConnection.lf:14]:
```suggestion
        let time =util.getElapsedLogicalTime();
```

File: test/TypeScript/src/PhysicalConnection.lf [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        console.log("Received event at logical time: " +  time);
+        if (time.isEqualTo(0)) {
```
Comment (lhstrh) [inline test/TypeScript/src/PhysicalConnection.lf:16]:
```suggestion
        if (time.isZero()) {
```

**Comment by lhstrh** [comment]
It's because the GitHub Action from ros-tools was broken. I submitted a PR with them and it is fixed now. Updating the branch and rerunning should do the trick.

**Comment by lhstrh** [comment]
The LSP tests are also failing in `master`, but that's being worked on...

**Pull request merged**